configparser/parser.go

```diff
@@ -0,0 +1,43 @@
+package configparser
+
+import (
+	"encoding/json"
+	"errors"
+
+	"gopkg.in/yaml.v3"
+)
+
+type extension string
+
+const (
+	extYML  extension = ".yml"
+	extYAML extension = ".yaml"
+	extJSON extension = ".json"
+)
```

Comment:
It's an enum. Enums are a well-suited pattern for a set of limited known values. They provide a better visibility on what values are accepted (rather than having to look inside the code in a switch statement) and self-documenting typing for function parameters: an extension is not any string.
The fact the compiler does not enforce that typing is not a sufficient reason to leave the other benefits.

`default` case *does* happen in case you provide a file with an invalid extension.

Reply:
Can't you assert the type is right when you convert? If no then i was wrong on the `default` case. It thought it was what was going on.

Reply:
The typing doesn't even matter here: at runtime the switch tries to match a value. If that value is `.bruvbzo`, then it's none of the cases, then its `default.`
Type safety can't do much at runtime, at this point it's error handling.

Reply:
Ok then mb